core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityTracker.scala

```diff
@@ -37,7 +38,8 @@ class CardinalityTracker(ref: DatasetRef,
                          shardKeyLen: Int,
                          defaultChildrenQuota: Seq[Int],
                          val store: CardinalityStore,
-                         quotaExceededProtocol: QuotaExceededProtocol = NoActionQuotaProtocol) extends StrictLogging {
+                         quotaExceededProtocol: QuotaExceededProtocol = NoActionQuotaProtocol,
+                         dsCardinalityMapFlushCount: Int = 5000) extends StrictLogging {
```

Comment:
1. Rename to `flushThreshold`` which can be ``Option``, if present will flush using that threshold or continue the existing path (the one used in raw)
2. We are leaking the cluster info in this class. It is ok to have flush thresholds but no references to DS or any cluster in this class, which includes repurposing the existing method to change behavior with and without flush threshold.

Reply:
Thank you :) Agree with you to avoid leaking of cluster info, so renaming the functions and variables. I do have some follow up questions, will follow up offline on those.